src/webgpu/shader/validation/shader_io/size.spec.ts

```diff
@@ -210,3 +210,21 @@ g.test('size_non_struct')
 
     t.expectCompileResult(data.pass, code);
   });
+
+g.test('size_creation_fixed_footprint')
+  .desc(`Test that @size is only valid on types that have creation-fixed footprint.`)
```

Comment:
Should this be iterating through a number of examples/types?

Reply:
I believe that a runtime-sized array is the only non-creation-fixed-footprint type that is valid inside a structure.